components/org.wso2.carbon.identity.api.server.authenticators/org.wso2.carbon.identity.api.server.authenticators.v1/src/main/java/org/wso2/carbon/identity/api/server/authenticators/v1/core/ServerAuthenticatorManagementService.java

```diff
@@ -0,0 +1,856 @@
+import org.wso2.carbon.identity.core.model.FilterTreeBuilder;
+import org.wso2.carbon.identity.core.model.Node;
+import org.wso2.carbon.identity.core.model.OperationNode;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementClientException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementServerException;
+import org.wso2.carbon.idp.mgt.model.IdpSearchResult;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.common.Util.base64URLEncode;
+
+/**
+ * Call internal osgi services to perform server authenticators related operations.
+ */
+public class ServerAuthenticatorManagementService {
+
+    private static final Log log = LogFactory.getLog(ServerAuthenticatorManagementService.class);
+
+    /**
+     * Retrieves the list of available authenticators.
+     *
+     * @param filter The filter string.
+     * @param limit  The items per page. **Not supported at the moment.**
+     * @param offset The offset to be used with the limit parameter. **Not supported at the moment.**
+     * @return The list of authenticators
+     */
+    public List<Authenticator> getAuthenticators(String filter, Integer limit, Integer offset) {
+
+        handleNotImplementedCapabilities(limit, offset);
+
+        try {
+            String filterAuthenticatorName = null;
+            String filterOperationForName = null;
+            ArrayList<String> filterTagsList = null;
+            int maximumItemPerPage = IdentityUtil.getMaximumItemPerPage();
+            if (StringUtils.isNotBlank(filter)) {
+                List<ExpressionNode> expressionNodes = getExpressionNodesForAuthenticator(filter);
+                if (CollectionUtils.isNotEmpty(expressionNodes)) {
+                    NameFilter nameFilter = getFilterAuthenticatorNameAndOperation(expressionNodes);
+                    if (nameFilter != null) {
+                        filterAuthenticatorName = nameFilter.getName();
+                        filterOperationForName = nameFilter.getOperation();
+                    }
+                    filterTagsList = getFilterTagsList(expressionNodes);
+                }
+            }
+
+            LocalAuthenticatorConfig[] localAuthenticatorConfigs = AuthenticatorsServiceHolder.getInstance()
+                    .getApplicationManagementService().getAllLocalAuthenticators(ContextLoader
+                            .getTenantDomainFromContext());
+
+            int localAuthenticatorsCount = localAuthenticatorConfigs.length;
+            RequestPathAuthenticatorConfig[] requestPathAuthenticatorConfigs = new RequestPathAuthenticatorConfig[0];
+            if (localAuthenticatorsCount < maximumItemPerPage) {
```

Comment:
Why we need this check to get the requestPathAuthenticatorConfigs? Shall we add a comment by explaining the logic?

Reply:
fixed with e0d117b57a31fb8167d880e7dfc6c3f603702ff7